independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/utils/BuildToolHelper.java

```diff
@@ -0,0 +1,80 @@
+package io.quarkus.bootstrap.utils;
+
+import static io.quarkus.bootstrap.util.QuarkusModelHelper.DEVMODE_REQUIRED_TASKS;
+
+import io.quarkus.bootstrap.resolver.AppModelResolverException;
+import io.quarkus.bootstrap.resolver.QuarkusGradleModelFactory;
+import io.quarkus.bootstrap.resolver.model.QuarkusModel;
+import io.quarkus.bootstrap.util.QuarkusModelHelper;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+/**
+ * Helper class used to expose build tool used by the project
+ */
+public class BuildToolHelper {
+
+    public enum BuildTool {
+        MAVEN("pom.xml"),
+        GRADLE("build.gradle");
+
+        private final String buildFile;
+
+        BuildTool(String buildFile) {
+            this.buildFile = buildFile;
+        }
+
+        public String getBuildFile() {
+            return buildFile;
+        }
+    }
+
+    private BuildToolHelper() {
+
+    }
+
+    public static boolean isMavenProject(Path project) {
+        Path currentPath = project;
+        while (currentPath != null) {
+            if (Files.exists(currentPath.resolve(BuildTool.MAVEN.getBuildFile()))) {
+                return true;
+            }
+            currentPath = currentPath.getParent();
+        }
+        return false;
+    }
+
+    public static Path getBuildFile(Path project, BuildTool tool) {
+        Path currentPath = project;
+        while (currentPath != null) {
+            if (Files.exists(currentPath.resolve(tool.getBuildFile()))) {
+                return currentPath;
+            }
+            currentPath = currentPath.getParent();
+        }
+        return null;
+    }
+
+    public static QuarkusModel enableGradleAppModel(Path projectRoot, String mode)
+            throws IOException, AppModelResolverException {
+        if (isMavenProject(projectRoot)) {
```

Comment:
If projectRoot is actually the project root we shouldn't be walking up to the root of the FS looking for the pom.xml.